src/pages/student_detail/StudentDetail.tsx

```diff
@@ -1,29 +1,78 @@
 import {
-  HeaderProperty,
   InformationContainer,
   Label,
   MainContainer,
+  StudentNameContainer,
+  HeaderBoxes,
+  StudentStatusStyle,
+  IconStyles,
+  ButtonWrapper,
+  AllInformationContainer,
+  PaymentInfoContainer,
   NotFoundStudentMessage,
-  PaymentInfo,
-  PaymentsContainer,
-  PaymentsTable,
-  Row,
+  InfoBoxesContainer,
   StudentInfo,
+  EditIconStyles,
+  HeaderContainer,
+  StudentSituationStyle,
   StudentInfoContainer,
+  ButtonContainer,
   SubTitle,
-  TableBody,
-  TableHeader,
   Title,
 } from "./StudentDetail.styles";
-import { formatCurrency } from "../../utils/Formatter";
-import { useGetStudentByIdQuery } from "../../app/services/StudentService";
-import CheckIcon from "../../assets/check.webp";
-import { useLocation } from "react-router-dom";
+import { useLocation } from "react-router";
+import { useNavigate } from "react-router";
+import {
+  useActivateStudentMutation,
+  useDeleteStudentMutation,
+  useGetStudentByIdQuery,
+} from "../../app/services/StudentService";
+import StudentIcon from "../../assets/student-icon.svg";
+import BackIcon from "../../assets/back-icon.svg";
+import InfoIcon from "../../assets/info-icon.svg";
+import EditIcon from "../../assets/edit-icon.svg";
+import DesactivateIcon from "../../assets/desactivate-icon.svg";
+import Button from "../../components/button/Button";
+import { ConfirmDialog } from "../../components/confirm_dialog/ConfirmDialog";
+import { EditarAlumno, MisPlanes } from "../../routes/RoutesUtils";
+import toast from "react-hot-toast";
+import { useEffect, useState } from "react";
+import type { StudentDetailResponse } from "../../app/types/responses/StudentDetailResponse.type";
 
 const StudentDetail = () => {
-  const location = useLocation();
-  const { studentId } = location.state || {};
-  const { data: student, isError } = useGetStudentByIdQuery(studentId!);
+  const { studentId } = useLocation().state;
+  const { data: student, isError } = useGetStudentByIdQuery(studentId);
+  const [activateStudent] = useActivateStudentMutation();
+  const [desactivateStudent] = useDeleteStudentMutation();
+  const [showConfirm, setShowConfirm] = useState(false);
+  const navigate = useNavigate();
+  const [isActive, setIsActive] = useState(student?.status ?? false);
+  useEffect(() => {
+    setIsActive(student?.status ?? false);
+  }, [student]);
+  const handleOpenConfirm = () => {
+    setShowConfirm(true);
+  };
+
+  const handleConfirm = async () => {
+    try {
+      if (isActive) {
+        await desactivateStudent(studentId).unwrap();
+        toast.success("El alumno ha sido dado de baja.");
+      } else {
+        await activateStudent(studentId).unwrap();
+        toast.success("El alumno ha sido dado de alta.");
+      }
+      setIsActive(!isActive);
+      setShowConfirm(false);
+    } catch (error) {
+      toast.error("Hubo un problema al actualizar el alumno");
+    }
+  };
```

Comment:
Siguiendo todos los comentarios anteriores, esto podriamos refactorizarlo asi


```suggestion
  const handleConfirm = () => {
    student.isActive ? deactivateStudentAndShowMessage(studentId) : activateStudentAndShowMessage(studentId);
  };
```

donde los metodos mencionados serian estos

```
  const deactivateStudentAndShowMessage = (studentId: string) => {
    desactivateStudent(studentId)
      .unwrap()
      .then(() => {
        toast.success("El alumno ha sido dado de baja.")
        setShowConfirm(false);
      })
  }
```

```
  const activateStudentAndShowMessage = (studentId: string) => {
    activateStudent(studentId)
      .unwrap()
      .then(() => {
        toast.success("El alumno ha sido dado de alta.")
        setShowConfirm(false);
      });
  }
```

Ahora, si quisieramos que el modal se cierre siempre, sin importar si la peticion finalizo correctamente o no, podriamos cambiar el `setShowConfirm(false)` en el metodo `handleConfirm` despues de llamar alguna de las dos funciones